diff --git a/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java b/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java
index acebb8688b7f46ac74e421ea13b8321b06949df1..378c5a0201db1a27a3c1aa1540d467d192c88455 100644
--- a/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java
+++ b/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java
@@ -1,7 +1,5 @@
 package org.telegram.messenger;

-import androidx.media3.common.util.Consumer;
-
 import org.telegram.tgnet.ConnectionsManager;
 import org.telegram.tgnet.TLRPC;
 import org.telegram.tgnet.tl.TL_stories;
@@ -34,12 +32,12 @@ public class ChannelBoostsController {
         connectionsManager = ConnectionsManager.getInstance(currentAccount);
     }

-    public void getBoostsStats(long dialogId, Consumer<TL_stories.TL_premium_boostsStatus> consumer) {
+    public void getBoostsStats(long dialogId, Utilities.Callback<TL_stories.TL_premium_boostsStatus> consumer) {
         TL_stories.TL_premium_getBoostsStatus req = new TL_stories.TL_premium_getBoostsStatus();
         req.peer = messagesController.getInputPeer(dialogId);
         connectionsManager.sendRequest(req, (response, error) -> AndroidUtilities.runOnUIThread(() -> {
             if (response != null) {
-                consumer.accept((TL_stories.TL_premium_boostsStatus) response);
+                consumer.run((TL_stories.TL_premium_boostsStatus) response);
             } else {
                 BaseFragment fragment = LaunchActivity.getLastFragment();
                 if (error != null && fragment != null && "CHANNEL_PRIVATE".equals(error.text)) {
@@ -59,12 +57,12 @@ public class ChannelBoostsController {
                 } else {
                     BulletinFactory.global().showForError(error);
                 }
-                consumer.accept(null);
+                consumer.run(null);
             }
         }));
     }

-    public void userCanBoostChannel(long dialogId, TL_stories.TL_premium_boostsStatus boostsStatus, Consumer<CanApplyBoost> consumer) {
+    public void userCanBoostChannel(long dialogId, TL_stories.TL_premium_boostsStatus boostsStatus, Utilities.Callback<CanApplyBoost> consumer) {
         CanApplyBoost canApplyBoost = new CanApplyBoost();
         canApplyBoost.currentPeer = messagesController.getPeer(dialogId);
         canApplyBoost.currentDialogId = dialogId;
@@ -72,7 +70,7 @@ public class ChannelBoostsController {
         BoostRepository.getMyBoosts(myBoosts -> {
             canApplyBoost.isMaxLvl = boostsStatus.next_level_boosts <= 0;
             canApplyBoost.setMyBoosts(myBoosts);
-            consumer.accept(canApplyBoost);
+            consumer.run(canApplyBoost);
         }, error -> {
             if (error.text.startsWith("FLOOD_WAIT")) {
                 canApplyBoost.floodWait = Utilities.parseInt(error.text);
@@ -80,7 +78,7 @@ public class ChannelBoostsController {
                 canApplyBoost.empty = true;
             }
             canApplyBoost.canApply = false;
-            consumer.accept(canApplyBoost);
+            consumer.run(canApplyBoost);
         });
     }
