diff --git a/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java b/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java
index 34ced4af68bf327e7d61d4197ae63f4259b08773..12dc19d2b9ab56e4d67ff3f852727d910b568c03 100644
--- a/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java
@@ -10,12 +10,8 @@ import android.webkit.MimeTypeMap;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
-import androidx.annotation.OptIn;
 import androidx.collection.LongSparseArray;
 
-import androidx.media3.common.util.Consumer;
-import androidx.media3.common.util.UnstableApi;
-
 import org.telegram.SQLite.SQLiteCursor;
 import org.telegram.SQLite.SQLiteDatabase;
 import org.telegram.SQLite.SQLitePreparedStatement;
@@ -92,7 +88,6 @@ import java.util.Objects;
 import java.util.Set;
 import java.util.TreeSet;
 
-@OptIn(markerClass = UnstableApi.class)
 public class StoriesController {
 
     public final static int STATE_READ = 0;
@@ -1568,16 +1563,16 @@ public class StoriesController {
 
     LongSparseArray<TL_stories.StoryItem> resolvedStories = new LongSparseArray<>();
 
-    public void resolveStoryAlbumLink(long peerId, int storyAlbumId, Consumer<TL_stories.TL_storyAlbum> consumer) {
+    public void resolveStoryAlbumLink(long peerId, int storyAlbumId, Utilities.Callback<TL_stories.TL_storyAlbum> consumer) {
         StoriesCollections collections = getStoryAlbumsList(peerId, false);
         if (collections != null) {
             StoryAlbum album = collections.findById(storyAlbumId);
             if (album != null) {
-                consumer.accept(album.toTl());
+                consumer.run(album.toTl());
                 return;
             }
             if (collections.loaded) {
-                consumer.accept(null);
+                consumer.run(null);
                 return;
             }
         }
@@ -1589,22 +1584,22 @@ public class StoriesController {
                 TL_stories.TL_albums albums = ((TL_stories.TL_albums) res);
                 for (TL_stories.TL_storyAlbum album : albums.albums) {
                     if (album.album_id == storyAlbumId) {
-                        consumer.accept(album);
+                        consumer.run(album);
                         return;
                     }
                 }
             }
 
-            consumer.accept(null);
+            consumer.run(null);
         }));
     }
 
-    public void resolveLiveStoryLink(long peerId, Consumer<TL_stories.StoryItem> consumer) {
+    public void resolveLiveStoryLink(long peerId, Utilities.Callback<TL_stories.StoryItem> consumer) {
         TL_stories.PeerStories userStoriesLocal = getStories(peerId);
         if (userStoriesLocal != null) {
             for (int i = 0; i < userStoriesLocal.stories.size(); i++) {
                 if (userStoriesLocal.stories.get(i).media instanceof TLRPC.TL_messageMediaVideoStream && !(userStoriesLocal.stories.get(i) instanceof TL_stories.TL_storyItemSkipped)) {
-                    consumer.accept(userStoriesLocal.stories.get(i));
+                    consumer.run(userStoriesLocal.stories.get(i));
                     return;
                 }
             }
@@ -1612,7 +1607,7 @@ public class StoriesController {
         long hash = peerId + -93321425 << 12;
         TL_stories.StoryItem storyItem = resolvedStories.get(hash);
         if (storyItem != null) {
-            consumer.accept(storyItem);
+            consumer.run(storyItem);
             return;
         }
         final TL_stories.TL_stories_getPeerStories req = new TL_stories.TL_stories_getPeerStories();
@@ -1636,18 +1631,18 @@ public class StoriesController {
                             }
                         }
                     }
-                    consumer.accept(storyItem);
+                    consumer.run(storyItem);
                 });
             }
         });
     }
 
-    public void resolveStoryLink(long peerId, int storyId, Consumer<TL_stories.StoryItem> consumer) {
+    public void resolveStoryLink(long peerId, int storyId, Utilities.Callback<TL_stories.StoryItem> consumer) {
         TL_stories.PeerStories userStoriesLocal = getStories(peerId);
         if (userStoriesLocal != null) {
             for (int i = 0; i < userStoriesLocal.stories.size(); i++) {
                 if (userStoriesLocal.stories.get(i).id == storyId && !(userStoriesLocal.stories.get(i) instanceof TL_stories.TL_storyItemSkipped)) {
-                    consumer.accept(userStoriesLocal.stories.get(i));
+                    consumer.run(userStoriesLocal.stories.get(i));
                     return;
                 }
             }
@@ -1655,7 +1650,7 @@ public class StoriesController {
         long hash = peerId + storyId << 12;
         TL_stories.StoryItem storyItem = resolvedStories.get(hash);
         if (storyItem != null) {
-            consumer.accept(storyItem);
+            consumer.run(storyItem);
             return;
         }
         TL_stories.TL_stories_getStoriesByID stories = new TL_stories.TL_stories_getStoriesByID();
@@ -1674,7 +1669,7 @@ public class StoriesController {
                             resolvedStories.put(hash, storyItem = response.stories.get(0));
                         }
                     }
-                    consumer.accept(storyItem);
+                    consumer.run(storyItem);
                 });
             }
         });
@@ -4508,7 +4503,7 @@ public class StoriesController {
         return null;
     }
 
-    public void canSendStoryFor(long dialogId, Consumer<Boolean> consumer, boolean showLimitsBottomSheet, Theme.ResourcesProvider resourcesProvider) {
+    public void canSendStoryFor(long dialogId, Utilities.Callback<Boolean> consumer, boolean showLimitsBottomSheet, Theme.ResourcesProvider resourcesProvider) {
         final TL_stories.TL_stories_canSendStory tl_stories_canSendStory = new TL_stories.TL_stories_canSendStory();
         tl_stories_canSendStory.peer = MessagesController.getInstance(currentAccount).getInputPeer(dialogId);
         ConnectionsManager.getInstance(currentAccount).sendRequest(tl_stories_canSendStory, (res, err) -> AndroidUtilities.runOnUIThread(() -> {
@@ -4518,12 +4513,12 @@ public class StoriesController {
                         MessagesController messagesController = MessagesController.getInstance(currentAccount);
                         messagesController.getBoostsController().getBoostsStats(dialogId, boostsStatus -> {
                             if (boostsStatus == null) {
-                                consumer.accept(false);
+                                consumer.run(false);
                                 return;
                             }
                             messagesController.getBoostsController().userCanBoostChannel(dialogId, boostsStatus, canApplyBoost -> {
                                 if (canApplyBoost == null) {
-                                    consumer.accept(false);
+                                    consumer.run(false);
                                     return;
                                 }
                                 BaseFragment lastFragment = LaunchActivity.getLastFragment();
@@ -4545,12 +4540,12 @@ public class StoriesController {
                                     };
                                 }
                                 LimitReachedBottomSheet.openBoostsForPostingStories(lastFragment, dialogId, canApplyBoost, boostsStatus, runnable);
-                                consumer.accept(false);
+                                consumer.run(false);
                             });
-                            consumer.accept(false);
+                            consumer.run(false);
                         });
                     } else {
-                        consumer.accept(false);
+                        consumer.run(false);
                     }
                 } else if (err.text.startsWith("STORY_LIVE_ALREADY_")) {
                     BaseFragment lastFragment = LaunchActivity.getLastFragment();
@@ -4561,22 +4556,22 @@ public class StoriesController {
                             .setPositiveButton(getString(R.string.OK), null)
                             .show();
                     }
-                    consumer.accept(false);
+                    consumer.run(false);
                 } else if (err.text.equalsIgnoreCase("PREMIUM_ACCOUNT_REQUIRED")) {
                     BaseFragment lastFragment = LaunchActivity.getLastFragment();
                     if (showLimitsBottomSheet && lastFragment != null) {
                         lastFragment.showDialog(new PremiumFeatureBottomSheet(lastFragment, PremiumPreviewFragment.PREMIUM_FEATURE_STORIES, true));
                     }
-                    consumer.accept(false);
+                    consumer.run(false);
                 } else {
                     BulletinFactory bulletinFactory = BulletinFactory.global();
                     if (bulletinFactory != null) {
                         bulletinFactory.showForError(err);
                     }
-                    consumer.accept(false);
+                    consumer.run(false);
                 }
             } else {
-                consumer.accept(true);
+                consumer.run(true);
             }
         }), ConnectionsManager.RequestFlagDoNotWaitFloodWait);
     }
