diff --git a/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java b/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java
index d6c20be75ad2e276f046662f5ddc1ccccca6d29b..acebb8688b7f46ac74e421ea13b8321b06949df1 100644
--- a/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java
+++ b/TMessagesProj/src/main/java/org/telegram/messenger/ChannelBoostsController.java
@@ -1,6 +1,6 @@
 package org.telegram.messenger;
 
-import com.google.android.exoplayer2.util.Consumer;
+import androidx.media3.common.util.Consumer;
 
 import org.telegram.tgnet.ConnectionsManager;
 import org.telegram.tgnet.TLRPC;
