diff --git a/TMessagesProj/src/main/java/org/telegram/ui/ChannelColorActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/ChannelColorActivity.java
index 2f13495280f9c61099f0a6fd0b0c1b9743c43d82..db92e58dad3ec297379fd9092b189372e44f7597 100644
--- a/TMessagesProj/src/main/java/org/telegram/ui/ChannelColorActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/ChannelColorActivity.java
@@ -361,7 +361,7 @@ public class ChannelColorActivity extends BaseFragment implements NotificationCe
             }
         });
 
-        sunDrawable = new RLottieDrawable(R.raw.sun, "" + R.raw.sun, dp(28), dp(28), true, null);
+        sunDrawable = new RLottieDrawable(R.raw.sun, dp(28), dp(28), true, null);
         sunDrawable.setPlayInDirectionOfCustomEndFrame(true);
         if (!isDark) {
             sunDrawable.setCustomEndFrame(0);
