diff --git a/TMessagesProj/src/main/java/org/telegram/ui/ChannelCreateActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/ChannelCreateActivity.java
index 60ce96221e006dd54072b894acc189f8dd9a1d43..e4cfabc138be3c600390e9f83a15f043d44d9c73 100644
--- a/TMessagesProj/src/main/java/org/telegram/ui/ChannelCreateActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/ChannelCreateActivity.java
@@ -605,7 +605,7 @@ public class ChannelCreateActivity extends BaseFragment implements NotificationC
                 avatarEditor.playAnimation();
             });
 
-            cameraDrawable = new RLottieDrawable(R.raw.camera, "" + R.raw.camera, AndroidUtilities.dp(60), AndroidUtilities.dp(60), false, null);
+            cameraDrawable = new RLottieDrawable(R.raw.camera, AndroidUtilities.dp(60), AndroidUtilities.dp(60), false, null);
 
             avatarEditor = new RLottieImageView(context) {
                 @Override
