diff --git a/TMessagesProj/src/main/java/org/telegram/ui/ChannelWallpaperActivity.java b/TMessagesProj/src/main/java/org/telegram/ui/ChannelWallpaperActivity.java
index dc822a62d9dc9df8da0ce9fc3837733c305f6353..33e685dc49a5e80a22a82c9be17a515dfab6b52d 100644
--- a/TMessagesProj/src/main/java/org/telegram/ui/ChannelWallpaperActivity.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/ChannelWallpaperActivity.java
@@ -27,7 +27,6 @@ import androidx.recyclerview.widget.RecyclerView;
 import org.telegram.messenger.AndroidUtilities;
 import org.telegram.messenger.ChatObject;
 import org.telegram.messenger.ChatThemeController;
-import org.telegram.messenger.DialogObject;
 import org.telegram.messenger.LocaleController;
 import org.telegram.messenger.MessagesController;
 import org.telegram.messenger.R;
@@ -47,8 +46,6 @@ import org.telegram.ui.Components.LayoutHelper;
 import org.telegram.ui.Components.RLottieDrawable;
 import org.telegram.ui.Components.RecyclerListView;
 
-import java.util.Locale;
-
 public class ChannelWallpaperActivity extends BaseFragment {
 
     public final long dialogId;
@@ -126,7 +123,7 @@ public class ChannelWallpaperActivity extends BaseFragment {
             }
         });
 
-        sunDrawable = new RLottieDrawable(R.raw.sun, "" + R.raw.sun, dp(28), dp(28), true, null);
+        sunDrawable = new RLottieDrawable(R.raw.sun, dp(28), dp(28), true, null);
         sunDrawable.setPlayInDirectionOfCustomEndFrame(true);
         if (!isDark()) {
             sunDrawable.setCustomEndFrame(0);
