diff --git a/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java b/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java
index 36debb7462600e589aed1d11f7779b2569b11cfc..34ced4af68bf327e7d61d4197ae63f4259b08773 100644
--- a/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java
+++ b/TMessagesProj/src/main/java/org/telegram/ui/Stories/StoriesController.java
@@ -10,9 +10,11 @@ import android.webkit.MimeTypeMap;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.annotation.OptIn;
 import androidx.collection.LongSparseArray;
 
-import com.google.android.exoplayer2.util.Consumer;
+import androidx.media3.common.util.Consumer;
+import androidx.media3.common.util.UnstableApi;
 
 import org.telegram.SQLite.SQLiteCursor;
 import org.telegram.SQLite.SQLiteDatabase;
@@ -90,6 +92,7 @@ import java.util.Objects;
 import java.util.Set;
 import java.util.TreeSet;
 
+@OptIn(markerClass = UnstableApi.class)
 public class StoriesController {
 
     public final static int STATE_READ = 0;
