diff --git a/TMessagesProj_AppStandalone/src/main/java/org/telegram/ui/SMSStatsActivity.java b/TMessagesProj_AppStandalone/src/main/java/org/telegram/ui/SMSStatsActivity.java
index 6f0844735df92d0096eaa1bf1f77dac04029fb49..bd9503ea7d38d8687b6c02e5e0f03691adebc654 100644
--- a/TMessagesProj_AppStandalone/src/main/java/org/telegram/ui/SMSStatsActivity.java
+++ b/TMessagesProj_AppStandalone/src/main/java/org/telegram/ui/SMSStatsActivity.java
@@ -331,7 +331,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
         items.add(new Item(VIEW_TYPE_HEADER));
         items.add(new Item(VIEW_TYPE_TABLE));
         items.add(Item.asShadow(null));
-        items.add(Item.asButton(BUTTON_TERMS, R.drawable.menu_intro, LocaleController.getString(R.string.SmsToS)));
+        items.add(Item.asButton(BUTTON_TERMS, org.telegram.messenger.R.drawable.menu_intro, LocaleController.getString(R.string.SmsToS)));
         items.add(Item.asButton(BUTTON_PREMIUM, R.drawable.menu_premium_main, LocaleController.getString(R.string.SmsPremiumBenefits)));
         if (state == SMSJobController.STATE_JOINED && !SMSJobController.getInstance(currentAccount).journal.isEmpty()) {
             items.add(Item.asButton(BUTTON_HISTORY, R.drawable.menu_sms_history, LocaleController.getString(R.string.SmsHistory)).setError(SMSJobController.getInstance(currentAccount).hasError()));
@@ -339,7 +339,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
         if (state == SMSJobController.STATE_JOINED) {
             final int simsCount = SMSJobController.getInstance(currentAccount).simsCount();
             if (simsCount > 1 || simsCount == 1 && Build.VERSION.SDK_INT < 22) {
-                items.add(Item.asButton(BUTTON_SIM, R.drawable.menu_storage_path, LocaleController.getString(R.string.SmsActiveSim)));
+                items.add(Item.asButton(BUTTON_SIM, org.telegram.messenger.R.drawable.menu_storage_path, LocaleController.getString(R.string.SmsActiveSim)));
             }
         }
         items.add(Item.asShadow(null));
@@ -383,7 +383,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                             new AlertDialog.Builder(getContext(), getResourceProvider())
                                     .setTitle(LocaleController.getString(R.string.SmsNoSimTitle))
                                     .setMessage(AndroidUtilities.replaceTags(LocaleController.getString(R.string.SmsNoSimMessage)))
-                                    .setPositiveButton(LocaleController.getString(R.string.OK), null)
+                                    .setPositiveButton(LocaleController.getString(org.telegram.messenger.R.string.OK), null)
                                     .show();
                             return;
                         }
@@ -391,7 +391,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                             if (err != null) {
                                 BulletinFactory.showError(err);
                             } else if (res instanceof TLRPC.TL_boolFalse) {
-                                BulletinFactory.global().createErrorBulletin(LocaleController.getString(R.string.UnknownError)).show();
+                                BulletinFactory.global().createErrorBulletin(LocaleController.getString(org.telegram.messenger.R.string.UnknownError)).show();
                             } else {
                                 SMSJobController.getInstance(currentAccount).setState(SMSJobController.STATE_JOINED);
                                 SMSJobController.getInstance(currentAccount).loadStatus(true);
@@ -421,7 +421,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                 AlertDialog d = new AlertDialog.Builder(getContext(), getResourceProvider())
                     .setTitle(LocaleController.getString(R.string.SmsDeactivateTitle))
                     .setMessage(LocaleController.getString(R.string.SmsDeactivateMessage))
-                    .setPositiveButton(LocaleController.getString(R.string.VoipGroupLeave), (di, w) -> {
+                    .setPositiveButton(LocaleController.getString(org.telegram.messenger.R.string.VoipGroupLeave), (di, w) -> {
                         finishFragment();
                         if (SMSJobController.getInstance(currentAccount).getState() == SMSJobController.STATE_JOINED) {
                             AndroidUtilities.runOnUIThread(() -> {
@@ -431,7 +431,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                             SMSJobController.getInstance(currentAccount).setState(SMSJobController.STATE_NONE);
                         }
                     })
-                    .setNegativeButton(LocaleController.getString(R.string.Back), null)
+                    .setNegativeButton(LocaleController.getString(org.telegram.messenger.R.string.Back), null)
                     .setDimAlpha(0.5f)
                     .create();
                 showDialog(d);
@@ -494,7 +494,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
     private TextView errorChipTextView;
 
     private void updateHeader() {
-        limitPreviewView = new LimitPreviewView(getContext(), R.drawable.msg_limit_chats, 0, 0, resourceProvider);
+        limitPreviewView = new LimitPreviewView(getContext(), org.telegram.messenger.R.drawable.msg_limit_chats, 0, 0, resourceProvider);
         limitPreviewView.isStatistic = true;
         limitPreviewView.setDarkGradientProvider(this::setDarkGradientLocation);
         aboveTitleView = new FrameLayout(getContext()) {
@@ -511,7 +511,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
         errorChipTextView.setText(getString(R.string.SmsAirplaneMode));
         errorChipTextView.setCompoundDrawablePadding(dp(6));
         errorChipTextView.setGravity(Gravity.CENTER);
-        Drawable errorDrawable = getContext().getResources().getDrawable(R.drawable.list_warning_sign).mutate();
+        Drawable errorDrawable = getContext().getResources().getDrawable(org.telegram.messenger.R.drawable.list_warning_sign).mutate();
         errorDrawable.setBounds(0, 0, errorDrawable.getIntrinsicWidth(), errorDrawable.getIntrinsicHeight());
         errorDrawable.setColorFilter(new PorterDuffColorFilter(Color.WHITE, PorterDuff.Mode.SRC_IN));
         errorChipTextView.setCompoundDrawables(errorDrawable, null, null, null);
@@ -602,7 +602,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                     new AlertDialog.Builder(getContext(), getResourceProvider())
                             .setTitle(LocaleController.getString(R.string.SmsNoSimTitle))
                             .setMessage(AndroidUtilities.replaceTags(LocaleController.getString(R.string.SmsNoSimMessage)))
-                            .setPositiveButton(LocaleController.getString(R.string.OK), null)
+                            .setPositiveButton(LocaleController.getString(org.telegram.messenger.R.string.OK), null)
                             .show();
                     return;
                 }
@@ -610,7 +610,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                     if (err != null) {
                         BulletinFactory.showError(err);
                     } else if (res instanceof TLRPC.TL_boolFalse) {
-                        BulletinFactory.global().createErrorBulletin(LocaleController.getString(R.string.UnknownError)).show();
+                        BulletinFactory.global().createErrorBulletin(LocaleController.getString(org.telegram.messenger.R.string.UnknownError)).show();
                     } else {
                         SMSJobController.getInstance(currentAccount).setState(SMSJobController.STATE_JOINED);
                         SMSJobController.getInstance(currentAccount).loadStatus(true);
@@ -792,7 +792,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
             smsSentTextView.setText("" + (status == null ? 0 : status.total_sent), animated);
             smsRemainingTextView.setText("" + remainingNumber, animated);
             if (status == null) {
-                sentSinceDateTextView.setText(LocaleController.getString(R.string.None), animated);
+                sentSinceDateTextView.setText(LocaleController.getString(org.telegram.messenger.R.string.None), animated);
             } else {
                 String date = LocaleController.formatDateAudio(status.total_since, false);
                 if (date.length() > 0) {
@@ -808,7 +808,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
                 }
                 giftSinceDateTextView.setText(date, animated);
             } else {
-                giftSinceDateTextView.setText(LocaleController.getString(R.string.None), animated);
+                giftSinceDateTextView.setText(LocaleController.getString(org.telegram.messenger.R.string.None), animated);
             }
             SpannableString giftLink = new SpannableString(LocaleController.getString(R.string.SmsLastGiftLink));
             if (status != null && status.last_gift_slug != null) {
@@ -1070,7 +1070,7 @@ public class SMSStatsActivity extends GradientHeaderActivity implements Notifica
             button.setOnClickListener(v -> {
                 dismiss();
             });
-            button.setText(LocaleController.getString(R.string.Close), false);
+            button.setText(LocaleController.getString(org.telegram.messenger.R.string.Close), false);
             buttonContainer.addView(button, LayoutHelper.createFrame(LayoutHelper.MATCH_PARENT, 48, Gravity.CENTER_VERTICAL, 16, 0, 16, 0));
             View buttonShadow = new View(getContext());
             buttonShadow.setBackgroundColor(Theme.getColor(Theme.key_windowBackgroundGray));
