diff --git a/buildSrc/build.gradle.kts b/buildSrc/build.gradle.kts
index bf93ef1074b35a3fa9b8029b2bd297e5a1e7659c..4d43d9f9579812c4c97ea5692cd131377f50f5ec 100644
--- a/buildSrc/build.gradle.kts
+++ b/buildSrc/build.gradle.kts
@@ -5,9 +5,13 @@ plugins {
 
 gradlePlugin {
     plugins {
-        register("lottiePreParser") {
-            id = "org.telegram.lottie-meta"
-            implementationClass = "org.telegram.lottie.LottieMetaPlugin"
+        register("telegramBuildPlugin") {
+            id = "org.telegram.build-plugin"
+            implementationClass = "org.telegram.plugin.TelegramBuildPlugin"
+        }
+        register("telegramBuildAppPlugin") {
+            id = "org.telegram.build-app-plugin"
+            implementationClass = "org.telegram.plugin.TelegramBuildAppPlugin"
         }
         register("testGenerator") {
             id = "test-generator"
@@ -36,7 +40,7 @@ tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile>().configureEach
 
 dependencies {
     implementation(gradleApi())
-    implementation("com.android.tools.build:gradle:8.10.1")
+    implementation("com.android.tools.build:gradle:8.13.2")
 
     implementation("com.squareup.moshi:moshi:1.15.0")
     implementation("com.squareup.moshi:moshi-kotlin:1.15.0")
