diff --git a/buildSrc/src/main/kotlin/org/telegram/lottie/LottieMetaExtension.kt b/buildSrc/src/main/kotlin/org/telegram/lottie/LottieMetaExtension.kt
deleted file mode 100644
index 933c4310337c396ac251a711e843bb7362f2b3b6..0000000000000000000000000000000000000000
--- a/buildSrc/src/main/kotlin/org/telegram/lottie/LottieMetaExtension.kt
+++ /dev/null
@@ -1,11 +0,0 @@
-package org.telegram.lottie
-
-import org.gradle.api.provider.Property
-
-abstract class LottieMetaExtension {
-    // Target package for the generated class. Defaults to the variant namespace.
-    abstract val packageName: Property<String>
-
-    // Simple name of the generated class.
-    abstract val className: Property<String>
-}
