diff --git a/TMessagesProj_AppStandalone/src/main/java/org/telegram/messenger/SMSJobController.java b/TMessagesProj_AppStandalone/src/main/java/org/telegram/messenger/SMSJobController.java
index 7fbc00f4cffeac65bc13c24ac92496a7333ff700..5c82abc4539d4656e069da45cae20b5a46b2ff35 100644
--- a/TMessagesProj_AppStandalone/src/main/java/org/telegram/messenger/SMSJobController.java
+++ b/TMessagesProj_AppStandalone/src/main/java/org/telegram/messenger/SMSJobController.java
@@ -1,35 +1,23 @@
 package org.telegram.messenger;
 
 import android.Manifest;
-import android.annotation.SuppressLint;
 import android.app.Activity;
-import android.app.Notification;
 import android.app.PendingIntent;
-import android.content.BroadcastReceiver;
 import android.content.Context;
 import android.content.Intent;
-import android.content.IntentFilter;
 import android.content.SharedPreferences;
-import android.content.pm.PackageManager;
-import android.database.Cursor;
-import android.net.Uri;
 import android.os.Build;
-import android.telecom.PhoneAccountHandle;
-import android.telephony.CellInfo;
 import android.telephony.SmsManager;
 import android.telephony.SubscriptionInfo;
 import android.telephony.SubscriptionManager;
 import android.telephony.TelephonyManager;
 import android.text.TextUtils;
-import android.util.Log;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.RequiresApi;
 import androidx.annotation.RequiresPermission;
 
 import org.telegram.PhoneFormat.PhoneFormat;
-import org.telegram.messenger.web.R;
-import org.telegram.tgnet.AbstractSerializedData;
 import org.telegram.tgnet.ConnectionsManager;
 import org.telegram.tgnet.InputSerializedData;
 import org.telegram.tgnet.OutputSerializedData;
@@ -37,7 +25,6 @@ import org.telegram.tgnet.SerializedData;
 import org.telegram.tgnet.TLObject;
 import org.telegram.tgnet.TL_smsjobs;
 import org.telegram.tgnet.TLRPC;
-import org.telegram.ui.Components.Bulletin;
 import org.telegram.ui.Components.BulletinFactory;
 import org.telegram.ui.LaunchActivity;
 import org.telegram.ui.SMSSubscribeSheet;
@@ -49,7 +36,6 @@ import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Map;
 
 public class SMSJobController implements NotificationCenter.NotificationCenterDelegate {
 
@@ -230,7 +216,7 @@ public class SMSJobController implements NotificationCenter.NotificationCenterDe
                 if (err != null) {
                     BulletinFactory.showError(err);
                 } else if (res instanceof TLRPC.TL_boolFalse) {
-                    BulletinFactory.global().createErrorBulletin(LocaleController.getString(R.string.UnknownError)).show();
+                    BulletinFactory.global().createErrorBulletin(LocaleController.getString(org.telegram.messenger.R.string.UnknownError)).show();
                 } else {
                     setState(SMSJobController.STATE_JOINED);
                     loadStatus(true);
@@ -820,7 +806,7 @@ public class SMSJobController implements NotificationCenter.NotificationCenterDe
             if (err != null) {
                 BulletinFactory.showError(err);
             } else if (res instanceof TLRPC.TL_boolFalse) {
-                BulletinFactory.global().createErrorBulletin(LocaleController.getString(R.string.UnknownError)).show();
+                BulletinFactory.global().createErrorBulletin(LocaleController.getString(org.telegram.messenger.R.string.UnknownError)).show();
             } else {
                 SMSJobController.getInstance(currentAccount).loadStatus(true);
                 SMSJobController.getInstance(currentAccount).checkIsEligible(true, null);
